crates/stages/src/pipeline.rs

```diff
@@ -116,6 +129,12 @@ impl<DB: Database> Pipeline<DB> {
         self
     }
 
+    /// Set a [SyncStateUpdater].
+    pub fn set_sync_state_updater(mut self, updater: U) -> Self {
```

Comment:
since this is not a setter
```suggestion
    pub fn with_sync_state_updater(mut self, updater: U) -> Self {
```